Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceRenderer.swift

```diff
@@ -36,33 +40,46 @@ internal class ExperienceRenderer: ExperienceRendering {
         self.analyticsObserver = ExperienceStateMachine.AnalyticsObserver(container: container)
     }
 
-    func show(experience: Experience, priority: RenderPriority, published: Bool, completion: ((Result<Void, Error>) -> Void)?) {
+    func show(experience: ExperienceData, completion: ((Result<Void, Error>) -> Void)?) {
         guard Thread.isMainThread else {
             DispatchQueue.main.async {
-                self.show(experience: experience, priority: priority, published: published, completion: completion)
+                self.show(experience: experience, completion: completion)
             }
             return
         }
 
-        if priority == .normal && stateMachine.state != .idling {
+        if !(experience.experiment?.shouldExecute() ?? true) {
```

Comment:
Nit, but I think a `guard` here is makes it immediately clearer that we have to return early and makes it slightly easier than thinking about `if !shouldExecute`.